### PR TITLE
Deny-MgmtPorts-From-Internet does not always block NSG creation

### LINKED ISSUE #1357: Deny-MgmtPorts-From-Internet does not always block NSG creation
**Describe the bug**
Policy is set to Deny and Enforce. We want to block any NSGs from being exposed to the internet on management ports.
If you try and add a rule to an existing NSG, it is blocked as expected.
If you deploy a new NSG with the rule defined, it bypasses the policy and lets you create the NSG. The policy then flags this rule, but it should be blocked.
This happens via the Azure Portal or via Bicep.

**Steps to reproduce**
This bicep script is blocked as expected, because the rules are split from the NSG creation.
```
resource testNsg 'Microsoft.Network/networkSecurityGroups@2022-07-01' = {
  name: 'nsg-test2'
  location: 'westus2'
}

resource testNsgRules 'Microsoft.Network/networkSecurityGroups/securityRules@2022-07-01' = {
  name: 'RDP'
  parent: testNsg
  properties: {
    access: 'Allow'
    description: 'Allow RDP'
    destinationAddressPrefix: '*'
    destinationPortRange: '3389'
    direction: 'Inbound'
    priority: 100
    protocol: '*'
    sourceAddressPrefix: '*'
    sourcePortRange: '*'
  }
}
```
If you instead put the rules in the initial creation, it creates the NSG successfully and bypasses the policy.
```
resource testNsg 'Microsoft.Network/networkSecurityGroups@2022-07-01' = {
  name: 'nsg-test'
  location: 'westus2'
  properties: {
    securityRules: [
      {
        name: 'RDP'
        properties: {
          access: 'Allow'
          description: 'Allow RDP'
          destinationAddressPrefix: '*'
          destinationPortRange: '3389'
          direction: 'Inbound'
          priority: 100
          protocol: '*'
          sourceAddressPrefix: '*'
          sourcePortRange: '*'
        }
      }
    ]
  }
}
```

**Screenshots**
You can see the NSG was created and bypassed the policy.
image

### PR DESCRIPTION
**Pull request by DavidRobson**
## Overview/Summary

Fixes a bug in [Management port access from the Internet should be blocked](https://www.azadvertizer.net/azpolicyadvertizer/Deny-MgmtPorts-From-Internet.html) when set to enforce in deny mode. If you deployed a resource that defined network security group rules in the properties of the network security group, it would let you create the rule anyway even though it violated the policy.

Closes #1357 

## This PR fixes/adds/changes/removes

1. Updated [Management port access from the Internet should be blocked](https://www.azadvertizer.net/azpolicyadvertizer/Deny-MgmtPorts-From-Internet.html)

### Breaking Changes

1. If somebody had deployed this policy in enforce mode, they could still deploy some resources. They will no longer be able to - but this should be the correct behavior now.
2. It now lists NSG rules as non compliant twice (once for the NSG, once for the rule). Not much we can do about this as if you don't apply the rule in both places, the enforcement doesn't work.

## Testing Evidence

See bicep files used for testing: [nsg.zip](https://github.com/Azure/Enterprise-Scale/files/11863822/nsg.zip)

In audit mode they all deploy, but are listed as not compliant. I also tested in deny mode to make sure they could not be deployed.
Screenshot 2023-06-26 144651

Also a VM can no longer be deployed with RDP/SSH open to public. Before this would deploy even though the policy should deny it.
VM create

### Testing URLs

N/A

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
  - https://github.com/Azure/Enterprise-Scale/pull/1359 @Springstone I have tested with this policy which seems to work for me. See which one we want to go with. 
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by DavidRobson** [comment]
I am just checking internally if I need to get some permission in regards to agreeing to the CLA. Hopefully shouldn't be a problem.

**Comment by Springstone** [comment]
> I am just checking internally if I need to get some permission in regards to agreeing to the CLA. Hopefully shouldn't be a problem.

Hope this has cleared internally. Please do confirm your org is happy with the conditions.
And again, many thanks for putting in the time and effort. This is one of our more complex policies and you're helping by making it better!

**Comment by DavidRobson** [comment]
@microsoft-github-policy-service agree company="Quest Software"

**Comment by DavidRobson** [comment]
@Springstone I agreed to the CLA above, although I couldn't get it to actually tag the bot for some reason? Is that OK like it is?

Thanks for doing additional testing always better to do more. I haven't rolled this out company wide yet as it takes a while to get all the approvals but assuming your testing is OK I can try and roll it out next week. We currently have the version on the main branch rolled out company wide in deny mode (with enforcement on).

I notice there's a merge conflict on the auto generated file. How do you resolve this? I think manually merging that file isn't ideal as it's auto generated? I was thinking I might be best rolling back my branch to before the bot committed, merging main in then doing a force push but don't want to break anything? What is the normal strategy there?

**Comment by Springstone** [comment]
> @Springstone I agreed to the CLA above, although I couldn't get it to actually tag the bot for some reason? Is that OK like it is?
> 
> Thanks for doing additional testing always better to do more. I haven't rolled this out company wide yet as it takes a while to get all the approvals but assuming your testing is OK I can try and roll it out next week. We currently have the version on the main branch rolled out company wide in deny mode (with enforcement on).
> 
> I notice there's a merge conflict on the auto generated file. How do you resolve this? I think manually merging that file isn't ideal as it's auto generated? I was thinking I might be best rolling back my branch to before the bot committed, merging main in then doing a force push but don't want to break anything? What is the normal strategy there?

For the policies.json auto generated file, just accept all incoming changes on your branch to resolve the merge conflict. It is auto-generated, there is nothing needed to be done by you :)

**Comment by Springstone** [comment]
I've tested and it looks great! Looks like it is successfully blocking all combinations of port entry using blocked ports. If you could address the `policies.json` conflict, I'll prepare the merge into `main`.

Many thanks for your great contribution.

File: docs/wiki/Whats-new.md
Comment (Springstone) [inline docs/wiki/Whats-new.md:54]:
```suggestion
- Bug fix for [Management port access from the Internet should be blocked](https://www.azadvertizer.net/azpolicyadvertizer/Deny-MgmtPorts-From-Internet.html) not enforcing deny effect when a deployment includes rules defined in network security group properties (i.e., when specifying rules when creating the NSG).
```

Comment (Springstone) [reply docs/wiki/Whats-new.md:54]: Adding a little clarity.

**Review by Springstone — APPROVED** [review]
Looks good, please review Whats New comment and address the merge conflict so we can kick off final test and merge to `main`

**Review by Springstone — CHANGES_REQUESTED** [review]
Sorry, approved before the changes were submitted :)

**Comment by Springstone** [comment]
@DavidRobson nudge :)

**Review by Springstone — APPROVED** [review]
LGTM

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**
